core/annotation-processor/src/main/kotlin/com/codattle/annotation/processing/generator/DaoModelBuilderGenerator.kt

```diff
@@ -0,0 +1,163 @@
+package com.codattle.annotation.processing.generator
+
+import com.codattle.annotation.processing.annotation.GenerateDaoModelBuilder
+import com.google.auto.service.AutoService
+import com.squareup.kotlinpoet.*
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import java.io.File
+import javax.annotation.processing.*
+import javax.lang.model.SourceVersion
+import javax.lang.model.element.ElementKind
+import javax.lang.model.element.ExecutableElement
+import javax.lang.model.element.TypeElement
+
+@AutoService(Processor::class)
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@SupportedOptions(DaoModelBuilderGenerator.KAPT_KOTLIN_GENERATED_OPTION_NAME)
+class DaoModelBuilderGenerator : AbstractProcessor() {
+
+    companion object {
+        const val KAPT_KOTLIN_GENERATED_OPTION_NAME = "kapt.kotlin.generated"
+    }
+
+    private data class Parameter(val name: String, val type: TypeName, val nonNullable: Boolean)
+
+    override fun getSupportedAnnotationTypes(): MutableSet<String> {
+        return mutableSetOf(GenerateDaoModelBuilder::class.java.name)
```

Comment:
Why this set is mutable?

Reply:
This code interoperates with Java and Java expects that all Lists are mutable.